event/parsing/provide.go

```diff
@@ -0,0 +1,85 @@
+package parsing
+
+import (
+	"github.com/go-kit/kit/log"
+	"github.com/xmidt-org/arrange"
+	"github.com/xmidt-org/bascule/acquire"
+	"github.com/xmidt-org/glaukos/event/queue"
+	"github.com/xmidt-org/webpa-common/logging"
+	"go.uber.org/fx"
+)
+
+// CodexConfig determines the auth and address for connecting to the codex cluster.
+type CodexConfig struct {
+	Address string
+	Auth    AuthAcquirerConfig
+}
+
+type AuthAcquirerConfig struct {
+	JWT   acquire.RemoteBearerTokenAcquirerOptions
+	Basic string
+}
+
+// ParsersIn brings together all of the different types of parsers that glaukos uses.
+type ParsersIn struct {
+	fx.In
+	Parsers []queue.Parser `group:"parsers"`
+}
+
+// Provide bundles everything needed for setting up all of the event objects
+// for easier wiring into an uber fx application.
+func Provide() fx.Option {
+	return fx.Options(
+		ProvideEventMetrics(),
+		fx.Provide(
+			arrange.UnmarshalKey("codex", CodexConfig{}),
+			fx.Annotated{
+				Group: "parsers",
+				Target: func(in MetricsIn) queue.Parser {
+					return MetadataParser{
+						MetadataFields:        in.MetadataFields,
+						UnparsableEventsCount: in.UnparsableEventsCount,
+					}
+				},
+			},
+			fx.Annotated{
+				Group: "parsers",
+				Target: func(logger log.Logger, in MetricsIn, codexConfig CodexConfig) queue.Parser {
+					codexAuth, err := determineCodexTokenAcquirer(logger, codexConfig)
```

Comment:
This can be further up in the `fx.Provide()` function, rather than as a part of the parsers function.  Then the function takes an additional `codexAuth acquire.Acquirer` parameter, and uber fx deals with the returned error from `determineCodexTokenAcquirer()`.